src/lib/FreeC/Pass/ResolverPass.hs

```diff
@@ -746,6 +746,7 @@ instance Resolvable IR.Expr where
     alts'      <- mapM resolve alts
     exprType'  <- mapM resolve exprType
     return (IR.Case srcSpan scrutinee' alts' exprType')
+  -- Shadow variable patterns and recursively
```

Comment:
```suggestion

  -- Shadow variable patterns and resolve recursively.
```

Or even better: move this up to the rule for `IR.Lambda` and update the comment there.